### PR TITLE
fix sentry and gp errors

### PR DESCRIPTION
**Pull request by ice-orion**
* Fix ANR on android, related thread - https://github.com/software-mansion/react-native-reanimated/issues/3879
* Ignore all API errors (4** + 5**)
* Ignore error with `messaging().getToken()` on android without Google Services

### CODE REVIEW

**Review by ice-hector — DISMISSED** [review]

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts [hunk 1]
```diff
@@ -120,7 +120,9 @@ export function* updateDeviceMetadataSaga(action: Action) {
         systemVersion: DeviceInfo.getSystemVersion(),
         pushNotificationToken:
           hasPushPermissions && !clearDeviceMetadata
-            ? messaging().getToken()
+            ? messaging()
+                .getToken()
+                .catch(_ => '') // different errors might be throws due to lack of google services on device
```
Comment (ice-ares) [inline src/store/modules/Devices/sagas/updateDeviceMetadata.ts]: u should still log it, right?

File: src/store/rootSaga.ts [hunk 2]
```diff
@@ -81,9 +81,9 @@ export function* rootSaga(): SagaIterator {
  */
 const isUnexpectedError = (error: unknown) => {
   /**
-   * Client errors
+   * API errors
    */
-  if (isApi4xxError(error)) {
+  if (isApiError(error)) {
```
Comment (ice-ares) [inline src/store/rootSaga.ts:86]: thats not what this function does, 4xx are not unexpected. 

U're using this function for to check if u should log the error or not.

So create a new func : shouldLog and do that there.


or rename this one to that, if u`re not using anywhere else except for logging

File: src/api/client/index.ts [hunk 3]
```diff
@@ -137,15 +137,6 @@ export const isApiError = (
   );
 };
 
-export const isApi4xxError = (error: unknown): error is AxiosError => {
```
Comment (ice-ares) [inline src/api/client/index.ts:140]: don't we need this for business cases?

**Review by ice-atlas — DISMISSED** [review]

**Review by ice-hades — DISMISSED** [review]

File: src/api/client/index.ts (hunk 3 above)
Comment (ice-orion) [reply src/api/client/index.ts:140]: nope, there were no usages except the one that was removed

File: src/store/rootSaga.ts (hunk 2 above)
Comment (ice-orion) [reply src/store/rootSaga.ts:86]: renamed

**Review by ice-ares — DISMISSED** [review]

**Review by ice-hector — APPROVED** [review]

**Review by ice-ares — APPROVED** [review]

**Review by ice-atlas — APPROVED** [review]

**Pull request merged**